scripts/bootloader/keygen.py

```diff
@@ -11,6 +11,22 @@
 import sys
 
 
+def gen_key():
+    from hashlib import sha256
```

Comment:
Yes, if you provide a key instead of having it generated you won't need the hashlib pip package.

Reply:
But then we have a hidden dependency. Things might be working ok for a user up until they have it generated, then their script crashes on an import error.

IMO this should just become an absolute dependency so that users can run the scripts however they choose without unexpected failures.

Also, imports in functions slow down execution because the package contents will be imported every time the context is entered. In this case just a function, but it's not strictly necessary here and hides package requirements.

Reply:
hashlib is a native Python library. I'd prefer it was at the top for clarity.

Reply:
Good point, overlooked that it was native. I still advocate never importing in functions unless there is a strong reason to do so, and agree with @oyvindronningstad that it is more clear at the top, where we can see everything the script needs to do any of its tasks.